src/model/balance.js

```diff
@@ -7,64 +7,70 @@ const errors = require('../util/errors')
 const NotAcceptedError = errors.NotAcceptedError
 const InvalidFieldsError = errors.InvalidFieldsError
 
+const BALANCE_PREFIX = 'balance__'
+
 module.exports = class Balance extends EventEmitter {
   constructor (opts) {
     super()
 
     this._maximum = new BigNumber(opts.maximum)
-    this._store = opts.store
-    this._cached = null
+    this._balance = null
 
-    // all reserved DB key prefixes are 9 characters
-    // so it lines up with 'transfer_'
-    this._key = 'balance__'
+    this._key = BALANCE_PREFIX
+    this._store = opts.store
+    this._write = Promise.resolve()
```

Comment:
Each asynchronous write operation is attached after the previous via `.then()`, so the first operation of the chain should just be a null promise.